compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -380,16 +380,12 @@ pub fn eval_in_interpreter<'mir, 'tcx>(
         }
         Ok(mplace) => {
             // Since evaluation had no errors, validate the resulting constant.
-
-            // Temporarily allow access to the static_root_alloc_id for the purpose of validation.
-            let static_root_alloc_id = ecx.machine.static_root_alloc_id.take();
-            let validation = const_validate_mplace(&ecx, &mplace, cid);
-            ecx.machine.static_root_alloc_id = static_root_alloc_id;
```

Comment:
I am a bit torn about whether `before_alloc_read` should be suppressed during validation. OTOH it is nice that we can now remove this code here. On the other hand, now if validation does do a read from this static, it will see `Uninit` and show a surprising error rather than triggering a cycle error. I don't know if it is even possible to do that, it would require a `copy_op_transmute` from the static to somewhere else.

If validation passes, we'll still get the cycle error from the actual read part of `copy_op`.